ui/app/AppLayouts/Chat/views/ChatHeaderContentView.qml

```diff
@@ -0,0 +1,344 @@
+                                         chatDetails: chatContentModule.chatDetails
+                                     })
+                }
+
+                onEditCommunityChannel: {
+                    root.rootStore.editCommunityChannel(
+                                chatId,
+                                newName,
+                                newDescription,
+                                newEmoji,
+                                newColor,
+                                newCategory,
+                                channelPosition // TODO change this to the signal once it is modifiable
+                                )
+                }
+                onAddRemoveGroupMember: {
+                    loader.sourceComponent = contactsSelector
+                }
+                onFetchMoreMessages: {
+                    root.rootStore.messageStore.requestMoreMessages();
+                }
+                onLeaveGroup: {
+                    chatContentModule.leaveChat();
+                }
+                onUpdateGroupChatDetails: {
+                    root.rootStore.chatCommunitySectionModule.updateGroupChatDetails(
+                                chatId,
+                                groupName,
+                                groupColor,
+                                groupImage
+                                )
+                }
+            }
+        }
+
+        Rectangle {
+            height: 24
+            width: 1
+            color: Theme.palette.directColor7
+            Layout.alignment: Qt.AlignVCenter
+            visible: (menuButton.visible || membersButton.visible || searchButton.visible)
+        }
+    }
+
+    Keys.onEscapePressed: { loader.sourceComponent = statusChatInfoButton }
+
+    // Chat toolbar content option 1:
+    Component {
+        id: statusChatInfoButton
+
+        StatusChatInfoButton {
+            objectName: "chatInfoBtnInHeader"
+            width: Math.min(implicitWidth, parent.width)
+            title: chatContentModule? chatContentModule.chatDetails.name : ""
+            subTitle: {
+                if(!chatContentModule)
+                    return ""
+
+                // In some moment in future this should be part of the backend logic.
+                // (once we add transaltion on the backend side)
+                switch (chatContentModule.chatDetails.type) {
+                case Constants.chatType.oneToOne:
+                    return (chatContentModule.isMyContact(chatContentModule.chatDetails.id) ?
+                                qsTr("Contact") :
+                                qsTr("Not a contact"))
+                case Constants.chatType.publicChat:
+                    return qsTr("Public chat")
+                case Constants.chatType.privateGroupChat:
+                    let cnt = root.usersStore.usersModule.model.count
+                    if(cnt > 1) return qsTr("%1 members").arg(cnt);
+                    return qsTr("1 member");
```

Comment:
While we refactor this, please use a proper plural:
https://hackmd.io/@status-desktop/B1eOaf-nd/%2FjNzZnH8ZRiynrrXJ--XpmQ#Plurals

Reply:
This is existing code, however I don't get this, since we get the count number from the backend, I assume a standard number will be shown?

Reply:
But the wording won't be correct; in many languages "2 items" isn't the same as "6 items"

Reply:
done